ravenframework/CodeInterfaceClasses/Abce/AbceCodeInterface.py

```diff
@@ -0,0 +1,169 @@
+      @ In, inputFiles, list, List of input files (length of the list depends on the number of inputs have been added in the Step is running this code)
+      @ In, executable, string, executable name with absolute path of ABCE/run.py
+      @ In, clargs, dict, optional, dictionary containing the command-line flags the user can specify in the input (e.g. under the node < Code >< clargstype =0 input0arg =0 i0extension =0 .inp0/ >< /Code >)
+      @ Out, returnCommand, tuple, tuple containing the generated command. returnCommand[0] is the command to run the code (string), returnCommand[1] is the name of the output root
+    """
+    if clargs==None:
+      raise IOError('No input file was specified in clargs!')
+    #check for duplicate extension use
+    usedExts = list(ext[0][0] for ext in clargs['input'].values() if len(ext) != 0)
+    if len(usedExts) != len(set(usedExts)):
+      raise IOError('ABCEInterface cannot handle multiple input files with the same extension.  You may need to write your own interface.')
+    for inf in inputFiles:
+      ext = '.' + inf.getExt() if inf.getExt() is not None else ''
+      try:
+        usedExts.remove(ext)
+      except ValueError:
+        pass
+    if len(usedExts) != 0:
+      raise IOError('Input extension',','.join(usedExts),'listed in XML node Code, but not found in the list of Input of <Files>')
+
+    def findSettingIndex(inputFiles,ext):
+      """
+      Find the settings file and return its index in the inputFiles list.
+      @ In, inputFiles, list of InputFile objects
+      @ In, ext, string, extension of the settings.yml file
+      @ Out, index, int, index of the settings file in the inputFiles list
+      """
+      for index,inputFile in enumerate(inputFiles):
+        if inputFile.getBase() == 'settings' and inputFile.getExt() == ext:
+          return index,inputFile
+      raise IOError('No settings file with extension '+ext+' found!')
+
+    def setOutputDir(settingsFile):
+      """
+      Set the output directory in the settings file.
+      @ In, settingsFile, InputFile object, settings.yml file
+      @ Out, None
+      """
+      # the settings file is the settings.yml file the scenario name is in node
+      # simulation -> scenario_name
+      # the output directory is the directory of the settings file in subdirectory "outputs/$scenario_name"
+      # get the scenario name by reading the settings.yml file
+      scenarioName = None
+      with open(settingsFile.getAbsFile(),'r') as settings:
+        for line in settings:
+          if 'scenario_name' in line:
+            scenarioName = line.split(':')[1].strip()
+            # remove the double quotes
+            scenarioName = scenarioName.replace('"','')
+            break
+      self._outputDirectory = os.path.join(os.path.dirname(settingsFile.getAbsFile()),'outputs',scenarioName)
+      #print("_outputDirectory", self._outputDirectory, "settingsFile", settingsFile.getAbsFile(), "scenarioName", scenarioName)
```

Comment:
print statement can be removed; if it would be useful to keep for debugging, add a comment above it explaining how it can be useful to keep commented

Reply:
Removed.

Reply:
My bad. I removed this as it was no longer needed but looks like that wasn't pushed. Thanks for taking care of that.